### DISCUSSION TITLE
Add an option to the file interface to enable/disable specific options in the dropdown

### DISCUSSION THREAD

**Discussion by uebmaster** [directus/directus#6545] [Feature Requests]
It would be a good option with regard to security because currently being able to select a file from the library shows all the existing files in it, regardless of whether the files were uploaded by another user, which may contain sensitive information from other users.

It would be good to be able to limit the file source options so that you can disable the option to select file from library.

![image](https://user-images.githubusercontent.com/5565506/123678823-6cb5e800-d80c-11eb-8c4f-c42a45eda487.png)

**Comment by benhaynes** [comment]
You could always limit the files shown in the file library with the Role's permissions. https://docs.directus.io/reference/filter-rules/#filter-rules

```
{
	"owner": {
		"_eq": "$CURRENT_USER"
	}
}
```

**Reply by uebmaster** [reply]
I tried that filter before post the idea, it doesn't works, but making a second try i check the DB and saw the fields from files table and saw the column "uploaded_by", i compared the id and saw that correspond to the user id, then i've replaced in the filter "owner" by "uploaded_by" and works.

Thanks.

PD: Later i remembered that i saw a time ago the "uploaded_by" propierty in the docs xD (https://docs.directus.io/reference/api/system/files/#files)

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
